components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/workflow/SubscriptionCreationWorkflow.java

```
@@ -88,8 +89,12 @@ public WorkflowResponse completeWorkflow(WorkflowExecutor workflowExecutor) thro
         }
 
         //Add subscription to gateway
-        apiGateway.addAPISubscription(subscription);
         apiSubscriptionDAO.updateSubscriptionStatus(getWorkflowReference(), subscriptionState);
+        try {
+            apiGateway.addAPISubscription(subscription);
+        } catch (GatewayException e) {
+            log.error("Error occurred while adding subscription to gateway", e);
```

don;t we need to throw the error upwards?